### ISSUE TITLE
quantification error due to the experimental design

### ISSUE THREAD

**Issue by AmirWIS** [smith-chem-wisc/MetaMorpheus#2192]
Running a set of 21 samples, I got the following error: "Error reading experimental design file: Error: The experimental design did not contain the file(s): [_directory and file name_]. Skipping quantification."
As a result, MM generated the AllQuantifiedPproteinGroups, AllPSMs and AllPeptide files, but without intensity values. 
I tried processing the data on several computers, but got the same error message. 
Only when removing that sample from the set, the process has ended successfully. 
The data file is not corrupted. Using another data processing software, I did not have any issues. 
I am using MM version 0.0.320
The data file is QEP2_ZE_17202_TECPR_3_29092022.raw (line#19 in the experimental design)
The experimental design (screen capture) is attached.
Please let me know what you think and how to solve it.
Thanks,
Amir
![image](https://user-images.githubusercontent.com/93867179/196707068-2f8d92b6-9cc3-40e4-92ed-ef3b4ed2dca3.png)

**Comment by trishorts** [comment]
can you search that one file by itself successfully in metamorpheus?

**Comment by AmirWIS** [comment]
Yes. since it is one file, the search took only few minutes. It ended successfully.

**Comment by acesnik** [comment]
🤖 Sorry this sat so long. Your answer to trishorts' question — that the file searched fine on its own — is what makes it diagnosable now, so here is what was happening.

One of your 21 files failed calibration. That part was fine and expected; MetaMorpheus copies such a file through uncalibrated and carries on. The bug was in the experimental design it then rewrote. Calibration kept a list of the files it could not calibrate, and in the version you were running it added a **bare filename with no extension** to that list, while the code that rewrote the design checked the list for a **full path with extension**. Those can never be equal, so the check was always false and every row of the new design was rewritten to point at `<name>-calib.mzML` — including the row for the one file where no calibrated file had been written. Quantification then looked for a file that did not exist, reported the one name you saw, and skipped.

That accounts for the shape of what you saw: exactly one file named, that same file being the one whose removal fixed the run, and a single-file search succeeding because a run with nothing to reconcile never hits the mismatch.

It was fixed as part of the timsTOF `.d` support work in #2520 — the title looks unrelated, but that PR reworked how calibration tracks files it cannot calibrate, and the list now stores full paths on both sides. First release carrying it is **1.1.0** (June 2025). You reported this on 0.0.320, which was already a year old at the time, so an upgrade is the fix here rather than any change to your data or design.

Closing on that basis. One narrow condition worth knowing, since it is the case that would bring this back: the matching is now done by looking for the design's filename inside the recorded paths, so a set where one file's name is a prefix of another's — `sample_1` alongside `sample_10`, say — could still pair the wrong file. None of your 21 names are prefixes of each other, so it was not your problem, but if you ever see "did not contain the file(s)" again on a set with names like that, please reopen and say so.
